src/controllers/authController.js

```diff
@@ -1,2 +1,73 @@
-// eslint-disable-next-line
-console.log("Auth controller to be used");
+import token from '../../src/helpers/generateToken';
+import Users from '../../sequelize/models/User'
+import Tokens from '../../sequelize/models/Token'
+import { development } from "../../sequelize/config/config.js";
+import { Sequelize, where } from "sequelize";
+import bcrypt from 'bcrypt'
+let sequelize = new Sequelize(development);
+let User = Users(sequelize, Sequelize);
+let Token = Tokens(sequelize, Sequelize)
+//User login
+class AuthController{
+    async login(req, res) {
+
+      const user = await User.findOne({ where: {email: req.body.email }});
+      if(!user) return res.status(400).json({message: "Wrong email detected!"});
+     
+      const isPasswordMatch = await bcrypt.compare(req.body.password,user.password);
+      
+      if (isPasswordMatch)
+      {
+
+      
+                  // JSON WEB TOKEN FOR ATHENTICATING LOGIN
+
+        const newToken = token({id:user.id, role:user.roleId});
+        await Token.create({token:newToken,ownerId:user.id,status:"active"}
+        ).then(data =>{
+          const accessor = user.dataValues
+          console.log(accessor)
+          res.status(200).json({message:"A token for your session has been saved!",
+          user:{
+            id: accessor.id,
+            firstName:accessor.firstName,
+            lastName:accessor.lastName,
+            email:accessor.email,
+            createdAt: accessor.createdAt,
+            updatedAt:accessor.updatedAt,
+            role: accessor.roleId
+          },
+          token:data.token});
+        })    
+      } else {
+          res.status(200).json({
+              message: "Incorrect email or password"
+          })
+      }
+
+}
+
+async logout(req, res) {
+    
+  const token = req?.headers?.authorization || req?.headers['x-access-token'] || req?.params.token
+  
+  const splitedToken = token.split(' ')[1];
+  const tokenExist = await Token.findOne({where: {token:splitedToken}})
+  if (tokenExist){
+    const status = tokenExist.status
+    if(status==='active'){
+      await Token.update({status: 'expired'},{ where: {token:splitedToken}})
```

Comment:
Deleting Token here can be helpful in terms of saving time and memory

Reply:
That is a great idea sir!